### ISSUE TITLE
Feedback from previous milestones

### ISSUE THREAD

**Issue by seshafi** [UBC-MDS/DSCI-522-2425-team35-Heart_disease_diagnostic_machine#32]
Please compile milestone feedback in this thread.

**Comment by seshafi** [comment]
Milestone 1:

- [ ] Summary in README should include high-level interpretation of analysis findings, as well 1-2 sentences on what they might mean (at least at a high-level). [MAREK]
- [x] No Creative Commons license (for project report) was specified. [HUI]
- [x] abstract/summary does not clearly report the major findings. [MAREK]
- [x] some of your result tables should not be code outuput. and should be actual tables (THINK THIS HAS BEEN FIXED?)

**Comment by seshafi** [comment]
Milestone 2:

- [x] The run instructions in your Dockerfile could be merged as one complete command, otherwise it would constitute of multiple redundant layers [LONG]
- [x] It is unlucky, but your github action workflow contains a typo, hence your whole worfklow failed. Run sed -i "s|image: ***/dsci-522-2425-team35-heart_disease_diagnostic_machine:.*|image: ***/dsci-522-2425-team35-heart_disease_diagnostic_machine:latest|" docker-compose.yml
sed: can't read docker-compose.yml: No such file or directory You renamed it as Docker-compose.yml [LONG]
- [x] Your instruction was not clear on how to run the analysis, it ended up after pasting the Jupyter lab URL on the browser. Then, what's next? Where should one find the analysis notebook? And, how to reproduce the analysis? Also, how should one clone the repository? [SARAH - will go back and change this once we're done the makefile.]

**Comment by seshafi** [comment]
**Peer feedback**
Stephanie:
- [x] I think you may have accidentally forgot to remove a file that was used to test some of your code: data/raw/load_and_transform_data.ipynb! [HUI - also delete everything from data/raw]
- [x] It's great that you have a MIT license for your code! You may want to have another license for the non-code portions of your work (such as Creative Commons 4.0 License). You may also want to have an attribution to Tiffany, for the Docker-publish.yml. Notes on how to cite code from a project that was openly licensed can be found [here](https://ubc-dsci.github.io/reproducible-and-trustworthy-workflows-for-data-science/lectures/220-licenses-and-copyright.html#how-do-i-cite-the-code-i-used-from-another-project-that-was-openly-licensed). [HUI]
- [x] It's nice how most of your scripts have documentation, I especially like how I can understand what each script does from the docstrings under the main function! It would be an added bonus if you could include a comment on how to run the script, which would be a little more convenient to understand intended usage, instead of having to refer to the README.md frequently. It could be as simple as: # Usage: <terminal command> ([example](https://ubc-dsci.github.io/reproducible-and-trustworthy-workflows-for-data-science/lectures/150-scripts.html#our-first-r-script)). [add comment in docstring on how to run script - copy over from README] [MAREK]
- [x] Unfortunately, I was unable to launch your container on my laptop on the first try. After changing your docker-compose.yml to have "8888:8888" ports, I was able to launch your container! Maybe port 9999 does not exist on every machine. Otherwise, I was able to rerun your analysis! :D [UHHHHHH - Marek test]
- [x] Just to let you know, it seems like 3_eda.py is saving pairwise_relationships.png to your root after using the command from your README.md (python scripts/3_eda.py  --train data/processed/train_df.csv --write-to results).  [HUI]
- [x] Nice writing overall! Very small formatting things could be improved, such as capitalization of level 3 headers. [MAREK]

Shashank:
- [x] I noticed that there are a couple of import statements in the middle of the code. It doesn’t break anything, but I think it’s usually a good idea to keep all the imports at the top of the script. You can take a look at the [specific lines here](https://github.com/UBC-MDS/DSCI-522-2425-team35-Heart_disease_diagnostic_machine/blob/19ceaeddc4d3fdbd0effb7e1db7daa05c72c8751/scripts/2_data_split_validate.py#L124-L126) for an example.   [HUI - its script #2 btw]
- [x] I saw that each script has just one function - main(). It might be nice to break things down into smaller functions to make your code more modular. This will not only make it easier to manage but also make your code look a lot cleaner. [assigned]
- [x] I ran the container on localhost:9999, and everything worked perfectly on my end. One thing I’d suggest is adding a few print statements in your scripts. For example, messages like Loading data... or Data loaded successfully would be super helpful. It’d let you know the script is running as expected and help you spot where something went wrong if there’s an issue. [LONG]
- [x] There are a few[ unnecessary import statements](https://github.com/UBC-MDS/DSCI-522-2425-team35-Heart_disease_diagnostic_machine/blob/19ceaeddc4d3fdbd0effb7e1db7daa05c72c8751/scripts/5_evaluate.py#L14-L21) in the script [5_evaluate.py](https://github.com/UBC-MDS/DSCI-522-2425-team35-Heart_disease_diagnostic_machine/blob/19ceaeddc4d3fdbd0effb7e1db7daa05c72c8751/scripts/5_evaluate.py). These statements can safely be omitted as they are not being used anywhere in the script.  [HUI]


Thamer:
- [x] Better train / test split. Your test set seems to be too small. This is highly apparent in the final confusion matrix in your report which only contains 22 records. (I see you already mention this in the end of your report but maybe a different train/test split would help). [MAREK]
- [x] In the Introduction, "Heart disease is the leading cause of death worldwide" is a killer hook (pun intended) but I think such a big statement could use a citation. [SARAH]
- [x] Minor typos such as "numneric" and "patent" (instead of "patient"). I like to use a spellchecker extension for VS Code to check my spelling in my documentation and markdown files: https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker [SARAH]
- [x] In your EDA, you vaguely mention that "there are correlations in some variables" without mentioning where the correlations actually are. I think it's a missed opportunity to make the reader's job easier. [HUI]
- [x] In your EDA plots, the one for Thalassemia shows null in its bar chart even though I believe this should not be the case after data cleaning and validation. [HUI]
- [x] In you results, you mention that "Our final test results yielded a F1 score of 0.625 and there is a large discrepancy to the training F1 score (0.626). This indicates that the model is overfitting." but I think a 0.001 is hardly a "large discrepancy" and I wouldn't say that model is overfitting based on its F1 scores. [SARAH]
- [x] In your results and conclusion, I think you should be consistent in how you refer to your test scores rather than saying "0.833" and sometimes saying "72.7" percent. [SARAH]
- [x] "Pickle" should not be included in your environment.yml as it is already included in Python. And it causes an error when trying to install the environment. [LONG]
- [x] When running your code I received a few scary looking warnings with some of them repeating over and over. I think you should probably handle or suppress them. [LONG - I think this is in script 4 or maybe 5]

**Comment by MarekB19** [comment]
Self-feedback:
- [x] target classification - class 2,3 and 4 should be labeled as 1 instead of removed (and check if logreg still best) [SARAH (with support if needed)]
- [x] train test split (80/20 instead of 90/10) [MAREK]
- [x] output raw data file from initial cleaning [SARAH]
